### PR TITLE
Co deputies

### PR DESCRIPTION
**Pull request by jackgoodby**

### CODE REVIEW

File: src/AppBundle/Controller/ClientController.php [hunk 1]
```diff
@@ -90,4 +100,42 @@ public function addAction(Request $request)
...
+                $user->setCoDeputyClientConfirmed(true);
+                $this->getRestClient()->put('user/'.$user->getId(), $user);
+                return $this->redirect($this->generateUrl('homepage'));
+            } else {
+                $form->get('lastname')->addError(new FormError('We could not match the client last name'));
+                $form->get('caseNumber')->addError(new FormError('We could not match the client case number'));
+            }
+        }
+
+        return [ 'form' => $form->createView()
```
Comment (elvisciotti) [inline src/AppBundle/Controller/ClientController.php]: once merged we can run the `php-cs-fixed` with the config inside the repo to fix code syntax

File: src/AppBundle/Service/Redirector.php [hunk 2]
```diff
@@ -97,33 +97,50 @@ public function getFirstPageAfterLogin()
...
-     * @return array [route, options]
+     * @param EntityDir\User $user
+     * @param string $currentRoute
+     * @return bool|string
      */
-    private function getLayDeputyHomepage(EntityDir\User $user, $enabledLastAccessedUrl = false)
+    public function getCorrectRouteIfDifferent(EntityDir\User $user, $currentRoute)
     {
+        // Redirect to appropriate homepage
+        if (in_array($currentRoute, ['lay_home','odr_index'])){
```
Comment (elvisciotti) [inline src/AppBundle/Service/Redirector.php]: odr_index should be merged into lay_home. For further discussion when everyone is back

File: src/AppBundle/Controller/CoDeputyController.php [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                $selfRegisterData->setCaseNumber($form['clientCaseNumber']->getData());
+
+                // validate against casRec
+                try {
+                    $this->getRestClient()->apiCall('post', 'selfregister/verifycodeputy', $selfRegisterData, 'array', [], false);
+                    $user->setCoDeputyClientConfirmed(true);
+                    $this->getRestClient()->put('user/' . $user->getId(), $user);
+                    return $this->redirect($this->generateUrl('homepage'));
+                } catch (\Exception $e) {
+                    $translator = $this->get('translator');
```
Comment (elvisciotti) [inline src/AppBundle/Controller/CoDeputyController.php]: wondering if it's worth moving this logic in a common place, as it's used by the self register controller too..

File: src/AppBundle/Controller/AbstractController.php [hunk 4]
```diff
@@ -42,11 +42,12 @@ protected function getUserWithData(array $jmsGroups = array())
...
     protected function getFirstClient($groups = ['user', 'user-clients', 'client'])
     {
-        $user = $this->getRestClient()->get('user/' . $this->getUser()->getId(), 'User', $groups);
+        $user = $this->getUserWithData($groups);
+
         /* @var $user User */
         $clients = $user->getClients();
 
-        return !empty($clients) ? $clients[0] : null;
+        return (is_array($clients) && !empty($clients[0]) && $clients[0] instanceof Client) ? $clients[0] : null;
```
Comment (slizzio) [inline src/AppBundle/Controller/AbstractController.php]: !empty($clients[0]) not needed. If its an array index 0 will always exist

**Pull request merged**